eZ/Publish/Core/IO/Tests/IOMetadataHandler/LegacyDFSClusterTest.php

```diff
@@ -212,19 +212,28 @@ public function testExistsNot()
 
     public function testDeletedirectory()
     {
+        $this->urlDecoratorMock
+            ->expects($this->once())
+            ->method('decorate')
+            ->will($this->returnValue('prefix/images/_alias/subfolder'));
+
         $statement = $this->createDbalStatementMock();
         $statement
             ->expects($this->once())
             ->method('bindValue')
-            ->with(1, 'folder/subfolder/%');
+            ->with(1, 'prefix/images/\_alias/subfolder/%');
+        $statement
+            ->expects($this->once())
+            ->method('execute')
+            ->will($this->returnValue(true));
```

Comment:
Taking into account my suggestion about QueryBuilder, this mock could look like:
```php

        $queryBuilderMock = $this
            ->getMockBuilder(QueryBuilder::class)
            ->disableOriginalConstructor()
            ->getMock();

        $this->dbalMock
            ->expects($this->once())
            ->method('createQueryBuilder')
            ->willReturn($queryBuilderMock);

        $queryBuilderMock->expects($this->at(0))
            ->method('delete')
            ->with('ezdfsfile')
            ->willReturn($queryBuilderMock);

        $queryBuilderMock->expects($this->at(1))
            ->method('where')
            ->with('name LIKE :spiPath ESCAPE :esc')
            ->willReturn($queryBuilderMock);

        $queryBuilderMock->expects($this->at(2))
            ->method('setParameter')
            ->with(':esc', '\\')
            ->willReturn($queryBuilderMock);

        $queryBuilderMock->expects($this->at(3))
            ->method('setParameter')
            ->with(':spiPath', 'prefix/images/\_alias/subfolder/%')
            ->willReturn($queryBuilderMock);

        $queryBuilderMock->expects($this->once())
            ->method('execute')
            ->willReturn(1);
```